lib/skate/detours/detours.ex

```diff
@@ -55,6 +56,31 @@ defmodule Skate.Detours.Detours do
     |> Repo.insert()
   end
 
+  @doc """
+  Creates a detour given a user id & detour id.
+  """
+  def create_detour_for_user(user_id, uuid, attrs \\ %{}) do
+    user = User.get_by_id!(user_id)
+
+    %Detour{
+      author: user,
+      uuid: uuid
+    }
+    |> Detour.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  Update or create a detour given a user id & detour id.
+  """
+  def update_or_create_detour_for_user(user_id, uuid, attrs \\ %{}) do
+    case get_detour_by_uuid!(uuid) do
+      # If there isn't a detour with this uuid, create one.
+      nil -> create_detour_for_user(user_id, uuid, attrs)
+      existing_detour -> update_detour(existing_detour, attrs)
+    end
```

Comment:
can we use [`upserts`](https://hexdocs.pm/ecto/constraints-and-upserts.html#upserts) instead of creating this manually?

Reply:
Hmmm I'm not sure. After removing the uuid from the db and using the `id` column as the unique identifier, the uuid coming from the frontend is either null (indicating the detour hasn't yet been added to the db) or it's a reference to the id of an existing detour. 

If the uuid was always non-null from the frontend, I see I could use upsert like in [Skate.Settings.User line 51](https://github.com/mbta/skate/blob/hp/test-branch-saving-to-db/lib/skate/settings/user.ex#L51).